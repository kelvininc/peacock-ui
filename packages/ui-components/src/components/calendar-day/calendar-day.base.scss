@import '../../assets/styles/globals';

:host {
	/**
	 * @prop --calendar-day-text-color-default: Text color when state is default.
	 * @prop --calendar-day-text-color-hover: Text color when state is hover.
	 * @prop --calendar-day-text-color-active: Text color when state is active.
	 * @prop --calendar-day-text-color-in-range: Text color when state is in-range.
	 * @prop --calendar-day-text-color-active-hover: Text color when state is active and cursor is on hover.
	 * @prop --calendar-day-text-color-disabled: Text color when state is disabled.
	 * @prop --calendar-day-background-color-default: Background color when state is default.
	 * @prop --calendar-day-background-color-hover: Background color when state is hover.
	 * @prop --calendar-day-background-color-active: Background color when state is active.
	 * @prop --calendar-day-background-color-in-range: Background color when state is in range.
	 * @prop --calendar-day-background-color-active-hover: Background color when state is active and cursor is on hover.
	 * @prop --calendar-day-background-color-disabled: Background color when state is disabled.
	 * @prop --calendar-day-width: The calendar container width.
	 * @prop --calendar-day-height: The calendar container height.
	 */

	--calendar-day-width: 32px;
	--calendar-day-height: 32px;
}

.calendar-day {
	@include kv-font-span-regular;
	width: var(--calendar-day-width);
	height: var(--calendar-day-height);

	display: flex;
	align-items: center;
	justify-content: center;

	background-color: var(--calendar-day-background-color-default);
	color: var(--calendar-day-text-color-default);

	user-select: none;
	cursor: pointer;

	&:hover:not(.calendar-day--disabled):not(.calendar-day--active) {
		background-color: var(--calendar-day-background-color-hover);
		color: var(--calendar-day-text-color-hover);

		border-radius: 2px;
	}

	&--in-range:not(.calendar-day--disabled):not(.calendar-day--active) {
		background-color: var(--calendar-day-background-color-in-range);
		color: var(--calendar-day-text-color-in-range);
	}

	&--active:not(.calendar-day--disabled) {
		background-color: var(--calendar-day-background-color-active);
		color: var(--calendar-day-text-color-active);

		&:hover {
			background-color: var(--calendar-day-background-color-active-hover);
			color: var(--calendar-day-text-color-active-hover);
		}
	}

	&--left-rounded:not(.calendar-day--disabled) {
		border-top-left-radius: 2px;
		border-bottom-left-radius: 2px;
	}

	&--right-rounded:not(.calendar-day--disabled) {
		border-top-right-radius: 2px;
		border-bottom-right-radius: 2px;
	}

	&--disabled {
		cursor: not-allowed;

		background-color: var(--calendar-day-background-color-disabled);
		color: var(--calendar-day-text-color-disabled);
	}
}
