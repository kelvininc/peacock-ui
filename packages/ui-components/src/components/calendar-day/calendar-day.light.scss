@import './calendar-day.base';
@import './calendar-day.night'; // TODO: Check with design team color palette for light theme
