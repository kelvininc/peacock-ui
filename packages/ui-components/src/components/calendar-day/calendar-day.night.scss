@import './calendar-day.base';

:host {
	--calendar-day-text-color-default: #{kv-color('neutral-4')};
	--calendar-day-text-color-hover: #{kv-color('neutral-0')};
	--calendar-day-text-color-active: #{kv-color('neutral-0')};
	--calendar-day-text-color-in-range: #{kv-color('neutral-4')};
	--calendar-day-text-color-active-hover: #{kv-color('neutral-0')};
	--calendar-day-text-color-disabled: #{kv-color('neutral-5')};
	--calendar-day-background-color-default: #{kv-color('neutral-7')};
	--calendar-day-background-color-hover: #{kv-color('neutral-6')};
	--calendar-day-background-color-active: #{kv-color('primary')};
	--calendar-day-background-color-in-range: #{kv-color('neutral-7')};
	--calendar-day-background-color-active-hover: #{kv-color('primary', 'dark')};
	--calendar-day-background-color-disabled: #{kv-color('neutral-3')};
}
