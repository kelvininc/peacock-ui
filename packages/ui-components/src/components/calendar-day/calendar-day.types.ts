import { EventEmitter } from '@stencil/core';

export interface ICalendarDay {
	/** (required) Calendar day */
	day: number;
	/** (optional) If `true`, the day is disabled */
	disabled?: boolean;
	/** (optional) If `true`, the day is with active */
	active?: boolean;
	/** (optional) If `true`, the day is between the two selected dates */
	isBetweenSelectedDates?: boolean;
	/** (optional) If `true`, the day is in range from a start date to the current hovered day */
	inHoverRange?: boolean;
	/** (optional) If `true` the day is the start or the end of the range */
	isEdge?: boolean;
	/** (optional) If `true`, the day is the day of `today`date */
	isToday?: boolean;
}

export interface ICalendarDayEvents {
	/** Emitted when day button is clicked */
	clickDay: EventEmitter<number>;
	/** Emitted when day button is mouse enter */
	mouseEnterDay: EventEmitter<number>;
	/** Emitted when day button is mouse leave */
	mouseLeaveDay: EventEmitter<number>;
}
