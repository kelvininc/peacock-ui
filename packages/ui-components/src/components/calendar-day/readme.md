# kv-calendar-day

<!-- Auto Generated Below -->


## Usage

### Angular

```html
<!-- Default -->
<kv-calendar-day [day]="12"></kv-calendar-day>

<!-- Active -->
<kv-calendar-day [day]="12" active></kv-calendar-day>

<!-- Disabled -->
<kv-calendar-day [day]="12" [disabled]="true"></kv-calendar-day>

<!-- In Range -->
<kv-calendar-day [day]="12" [inRange]="true"></kv-calendar-day>
```


### React

```tsx
import React from 'react';

import { KvCalendarDay } from '@kelvininc/react-ui-components';

export const KvCalendarDayExample: React.FC = () => (
	<>
		{/*-- Default --*/}
		<KvCalendarDay day={12} />

		{/*-- Active --*/}
		<KvCalendarDay day={12} active={true} />

		{/*-- Disabled --*/}
		<KvCalendarDay day={12} disabled={true} />

		{/*-- In Range --*/}
		<KvCalendarDay day={12} inRange={true} />
	</>
);
```



## Properties

| Property       | Attribute       | Description                                            | Type      | Default     |
| -------------- | --------------- | ------------------------------------------------------ | --------- | ----------- |
| `active`       | `active`        | (optional) If `true`, the day is with active style     | `boolean` | `false`     |
| `day`          | `day`           | (required) Calendar day                                | `number`  | `undefined` |
| `disabled`     | `disabled`      | (optional) If `true`, the day is disabled              | `boolean` | `false`     |
| `inRange`      | `in-range`      | (optional) If `true`, the day is with 'in-range' style | `boolean` | `false`     |
| `leftRounded`  | `left-rounded`  | (optional) If `true`, the day is left rounded style    | `boolean` | `false`     |
| `rightRounded` | `right-rounded` | (optional) If `true`, the day is right rounded style   | `boolean` | `false`     |


## Events

| Event           | Description                            | Type                            |
| --------------- | -------------------------------------- | ------------------------------- |
| `clickDay`      | Emitted when day button is clicked     | `CustomEvent<IClickDayEvent>`   |
| `mouseEnterDay` | Emitted when day button is mouse enter | `CustomEvent<IMouseEnterEvent>` |
| `mouseLeaveDay` | Emitted when day button is mouse leave | `CustomEvent<IMouseLeaveEvent>` |


## Shadow Parts

| Part              | Description               |
| ----------------- | ------------------------- |
| `"day-container"` | The day button container. |


## CSS Custom Properties

| Name                                           | Description                                                   |
| ---------------------------------------------- | ------------------------------------------------------------- |
| `--calendar-day-background-color-active`       | Background color when state is active.                        |
| `--calendar-day-background-color-active`       | Background color when state is active.                        |
| `--calendar-day-background-color-active-hover` | Background color when state is active and cursor is on hover. |
| `--calendar-day-background-color-active-hover` | Background color when state is active and cursor is on hover. |
| `--calendar-day-background-color-default`      | Background color when state is default.                       |
| `--calendar-day-background-color-default`      | Background color when state is default.                       |
| `--calendar-day-background-color-disabled`     | Background color when state is disabled.                      |
| `--calendar-day-background-color-disabled`     | Background color when state is disabled.                      |
| `--calendar-day-background-color-hover`        | Background color when state is hover.                         |
| `--calendar-day-background-color-hover`        | Background color when state is hover.                         |
| `--calendar-day-background-color-in-range`     | Background color when state is in range.                      |
| `--calendar-day-background-color-in-range`     | Background color when state is in range.                      |
| `--calendar-day-height`                        | The calendar container height.                                |
| `--calendar-day-height`                        | The calendar container height.                                |
| `--calendar-day-text-color-active`             | Text color when state is active.                              |
| `--calendar-day-text-color-active`             | Text color when state is active.                              |
| `--calendar-day-text-color-active-hover`       | Text color when state is active and cursor is on hover.       |
| `--calendar-day-text-color-active-hover`       | Text color when state is active and cursor is on hover.       |
| `--calendar-day-text-color-default`            | Text color when state is default.                             |
| `--calendar-day-text-color-default`            | Text color when state is default.                             |
| `--calendar-day-text-color-disabled`           | Text color when state is disabled.                            |
| `--calendar-day-text-color-disabled`           | Text color when state is disabled.                            |
| `--calendar-day-text-color-hover`              | Text color when state is hover.                               |
| `--calendar-day-text-color-hover`              | Text color when state is hover.                               |
| `--calendar-day-text-color-in-range`           | Text color when state is in-range.                            |
| `--calendar-day-text-color-in-range`           | Text color when state is in-range.                            |
| `--calendar-day-width`                         | The calendar container width.                                 |
| `--calendar-day-width`                         | The calendar container width.                                 |


## Dependencies

### Used by

 - [kv-calendar](../calendar)

### Graph
```mermaid
graph TD;
  kv-calendar --> kv-calendar-day
  style kv-calendar-day fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


