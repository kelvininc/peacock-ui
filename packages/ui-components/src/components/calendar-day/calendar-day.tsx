import { Component, Event, EventEmitter, Host, Prop, h } from '@stencil/core';
import { ICalendarDay, ICalendarDayEvents } from './calendar-day.types';

/**
 * @part day-container - The day button container.
 */
@Component({
	tag: 'kv-calendar-day',
	styleUrl: 'calendar-day.scss',
	shadow: true
})
export class KvCalendarDay implements ICalendarDay, ICalendarDayEvents {
	/** @inheritdoc */
	@Prop({ reflect: true }) day!: number;
	/** @inheritdoc */
	@Prop({ reflect: true }) disabled?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: true }) active?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: false }) inRange?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: false }) isRangeStartDate?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: false }) isRangeEndDate?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: false }) isToday?: boolean = false;
	/** @inheritdoc */
	@Prop({ reflect: false }) isBetweenSelectedDates?: boolean = false;

	/** @inheritdoc */
	@Event() clickDay: EventEmitter<number>;
	/** @inheritdoc */
	@Event() mouseEnterDay: EventEmitter<number>;
	/** @inheritdoc */
	@Event() mouseLeaveDay: EventEmitter<number>;

	private onClickDay = (): void => {
		if (this.disabled) {
			return;
		}

		this.clickDay.emit(this.day);
	};

	private onMouseEnterDay = (): void => {
		this.mouseEnterDay.emit(this.day);
	};

	private onMouseLeaveDay = (): void => {
		this.mouseLeaveDay.emit(this.day);
	};

	render() {
		return (
			<Host>
				<div
					onClick={this.onClickDay}
					class={{
						'calendar-day': true,
						'calendar-day--disabled': this.disabled,
						'calendar-day--active': this.active,
						'calendar-day--in-range': this.inRange,
						'calendar-day--today': this.isToday,
						'calendar-day--range-start': this.isRangeStartDate,
						'calendar-day--range-end': this.isRangeEndDate,
						'calendar-day--between-dates': this.isBetweenSelectedDates
					}}
					onMouseEnter={this.onMouseEnterDay}
					onMouseLeave={this.onMouseLeaveDay}
					part="day-container"
				>
					{this.day}
				</div>
			</Host>
		);
	}
}
