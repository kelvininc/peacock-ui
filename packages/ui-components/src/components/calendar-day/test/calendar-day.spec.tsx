import { SpecPage } from '@stencil/core/internal';
import { KvCalendarDay } from '../calendar-day';
import { newSpecPage } from '@stencil/core/testing';

describe('Calendar Day (unit tests)', () => {
	let page: SpecPage;
	let component: KvCalendarDay;

	describe('when uses default props', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvCalendarDay],
				html: '<kv-calendar-day day="12"></kv-calendar-day>'
			});
			component = page.rootInstance;
		});

		it('should match the snapshot', () => {
			expect(page.root).toMatchSnapshot();
		});

		it('should initialize `disabled` with false', () => {
			expect(component.disabled).toBe(false);
		});

		it('should initialize `active` with false', () => {
			expect(component.active).toBe(false);
		});

		it('should initialize `inRange` with false', () => {
			expect(component.inRange).toBe(false);
		});

		it('should initialize `inRangeStartDate` with false', () => {
			expect(component.isRangeStartDate).toBe(false);
		});

		it('should initialize `inRangeEndDate` with false', () => {
			expect(component.isRangeEndDate).toBe(false);
		});

		it('should initialize `isToday` with false', () => {
			expect(component.isToday).toBe(false);
		});

		it('should initialize `isBetweenSelectedDates` with false', () => {
			expect(component.isBetweenSelectedDates).toBe(false);
		});
	});
});
