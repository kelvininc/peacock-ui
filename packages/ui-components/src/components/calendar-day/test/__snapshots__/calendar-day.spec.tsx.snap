// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Day (unit tests) when uses default props should match the snapshot 1`] = `
<kv-calendar-day day="12">
  <mock:shadow-root>
    <div class="calendar-day" part="day-container">
      12
    </div>
  </mock:shadow-root>
</kv-calendar-day>
`;
